translator/translate/logs/util/filter_retention.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/translator"
+	"strings"
+)
+
+func CheckForConflictingRetentionSettings(logConfigs []interface{}, currPath string) []interface{} {
+	configMap := make(map[string]int)
+	for _, logConfig := range logConfigs {
+		if logConfigMap, ok := logConfig.(map[string]interface{}); ok {
+			// skip if retention is not set
+			if retention, ok := logConfigMap["retention_in_days"].(int); ok {
+				if logGroup, ok := logConfigMap["log_group_name"].(string); ok {
+					// if retention is 0, -1 or less it's either invalid or default
+					logGroup = strings.ToLower(logGroup)
+					if retention < 1 {
+						continue
+					}
```

Comment:
You could do:
```
if retention, ok := logConfigMap["retention_in_days"].(int); ok {
    if retention < 1 {
        continue
    }
    if logGroup, ok := logConfigMap["log_group_name"].(string); ok {
.
.
.
```
Move the retention value check immediately after getting the retention value.